Documentation/ContentElementRendering/Index.rst

```diff
@@ -0,0 +1,157 @@
+such an extension, no content would be rendered at all. The default
+templates provided by this extension can be overridden with site settings
+provided by Fluid Styled Content.
+
+..  contents::
+
+..  _content-element-rendering-settings:
+
+Use Settings to override template paths of Fluid Styled Content
+===============================================================
+
+Site settings can be saved both in the site configuration and in the site
+package extension.
+
+We will save the settings to the site package but use the settings editor to
+write the YAML for us
+
+Go to module :guilabel:`Site Management > Settings` and edit the settings of
+your site. Override the paths to the templates of Fluid Styled Content like this:
+
+..  figure:: /Images/SiteSettingsFsc.png
+    :alt: Screenshot demonstrating the site settings module
+
+    Override the templates of Fluid Styled Content
+
+If you would click :guilabel:`Save` now, the settings would be saved to your
+site configuration at :path:`config/sites/my-site/settings.yaml`. We however
+want to save the settings to the site set of our site package extension.
+
+Click the button :guilabel:`YAML export` to copy the configuration to your
+Clipboard instead, then save it to the following file:
+
+..  code-block:: yaml
+    :caption: packages/site_package/Configuration/Sets/SitePackage/settings.yaml
+
+    styles:
+      templates:
+        templateRootPath: 'EXT:site_package/Resources/Private/ContentElements/Templates'
+        partialRootPath: 'EXT:site_package/Resources/Private/ContentElements/Partials'
+        layoutRootPath: 'EXT:site_package/Resources/Private/ContentElements/Layouts'
+
+Then close the site settings without saving. You can now start overriding the
+templates:
+
+..  _content-element-rendering-menu-subpages:
+
+Override the "Menu of Subpages" template
+========================================
+
+On "Page 1" of the example data a content element of type "Subpages" was added.
+We now have to find out what this type is called in the database. The raw values
+saved to the database are displayed in the TYPO3 backend when the debug mode
+is activated:
+
+..  figure:: /Images/MenuContentElement.png
+    :alt: Screenshot demonstrating the backend debug mode for Content element "Subpages" on "Page 1"
+
+    You can now see that the Type (saved in field `CType`) is stored as `menu_subpages`
+
+Now we must find and copy the original template from Fluid Styled Content. TYPO3
+extensions are saved by their composer key, here :composer:`typo3/cms-fluid-styled-content`,
+into the folder :path:`vendor` during installation via Composer. You can find
+the files belonging to Fluid Styled Content in folder
+:path:`vendor/typo3/cms-fluid-styled-content` therefore. This folder is
+structured similarly to your site package extension and you can find the original
+templates in folder :path:`Resources/Private/Templates` here.
+
+The templates have the name of the `CType` in CamelCase. Copy file
```

Comment:
This is a very very simplified interpretation and does not respect the true mapping. Sure, it may work for TYPO3 fluid_styled_content, but it may change every time and you can not expect people to have same mapping understanding in other extensions.

On the one hand your explanation is very easy to understand and helpful, but on the other hand it's nothing to relate on for other extensions. Hmmm....

Example on how the real mapping happens:

```
tt_content.menu_recently_updated =< lib.contentElement
tt_content.menu_recently_updated {
    templateName = MenuRecentlyUpdated
```

Reply:
Ok, I added a restriction by conventions and templates of Fluid styled content.